### PR TITLE
commit

### PR DESCRIPTION
**Pull request by moran711**

### CODE REVIEW

**Comment by RevylaV** [comment]
Hope its done

**Comment by moran711** [comment]
Please make function with such functionality: 
`createValidator({ minlength: 2 }, 'code');`
That will return us `CODE_NOT_VALID` error message.

This function will accept options object with built-in mongo validators: 
- All SchemaTypes have the built-in `required `validator. The required validator uses the SchemaType's checkRequired() function to determine if the value satisfies the required validator.
- Numbers have `min `and `max `validators.
- Strings have `enum`, `match`, `minLength`, and `maxLength `validators.
For `validate` field we will pass our function with custom validation

**Comment by moran711** [comment]
I think it would be better to create function for error message generation, because we will avoid code duplication in our config file.

**Comment by RevylaV** [comment]
Final

File: error-messages/address.messages.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const COUNTRY_LENGTH_TO_LONG = 'COUNTRY_LENGTH_TO_LONG';
+const REGION_LENGTH_TO_SHORT = 'REGION_LENGTH_TO_SHORT';
+const REGION_LENGTH_TO_LONG = 'REGION_LENGTH_TO_LONG';
+const CITY_LENGTH_TO_SHORT = 'CITY_LENGTH_TO_SHORT';
+const CITY_LENGTH_TO_LONG = 'CITY_LENGTH_TO_LONG';
+const ZIPCODE_LENGTH_TO_SHORT = 'ZIPCODE_LENGTH_TO_SHORT';
+const ZIPCODE_LENGTH_TO_LONG = 'ZIPCODE_LENGTH_TO_LONG';
+const STREET_LENGTH_TO_SHORT = 'STREET_LENGTH_TO_SHORT';
+const STREET_LENGTH_TO_LONG = 'STREET_LENGTH_TO_LONG';
+const BUILDINGNUMBER_LENGTH_TO_SHORT = 'BUILDINGNUMBER_LENGTH_TO_SHORT';
```
Comment (moran711) [inline error-messages/address.messages.js:11]: Please change it to `BUILDING_NUMBER_LENGTH_TO_SHORT `

File: error-messages/address.messages.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const BUILDINGNUMBER_LENGTH_TO_LONG = 'BUILDINGNUMBER_LENGTH_TO_LONG';
```
Comment (moran711) [inline error-messages/address.messages.js:12]: Please change it to `BUILDING_NUMBER_LENGTH_TO_LONG `

File: error-messages/address.messages.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const COUNTRY_LENGTH_TO_SHORT = 'COUNTRY_LENGTH_TO_SHORT';
```
Comment (moran711) [inline error-messages/address.messages.js:1]: Typo in `TO`

File: models/Language.js [hunk 4]
```diff
@@ -1,7 +1,16 @@
 const mongoose = require('mongoose');
+const { INPUT_NOT_VALID } = require('../error-messages/common.messages');
 
 const languageSchema = new mongoose.Schema({
-  lang: String,
+  lang: {
+    type: String,
+    validate: {
+      validator: function(v) {
+        return /^[а-яїієґ0-9\s]+$/i.test(v) || /^[a-z0-9\s]+$/i.test(v);
```
Comment (moran711) [inline models/Language.js:9]: Please move this magic regexp to config file

File: modules/comment/common/ImageSet.js [hunk 5]
```diff
@@ -1,10 +1,26 @@
 const mongoose = require('mongoose');
 
 const imageSetSchema = new mongoose.Schema({
-  large: String,
-  medium: String,
-  small: String,
-  thumbnail: String,
+  large: {
+    type: String,
+    minlength: [2, 'error'],
```
Comment (moran711) [inline modules/comment/common/ImageSet.js:6]: Please move this error to config file and name it due to your changes

File: modules/comment/common/ImageSet.js [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+  small: {
+    type: String,
+    minlength: [2, 'error'],
+    maxlength: [20, 'error'],
+  },
+  thumbnail: {
+    type: String,
+    minlength: [2, 'error'],
+    maxlength: [20, 'error'],
+  },
```
Comment (moran711) [inline modules/comment/common/ImageSet.js:23]: same

File: modules/contact/contact.model.js [hunk 7]
```diff
@@ -1,19 +1,48 @@
...
+    required: [true, PHONE_NUMBER_IS_REQUIRED],
+  },
   openHours: [Language],
   address: [Language],
-  email: String,
+  email: {
+    type: String,
+    validate: {
+      validator: function(v) {
+        return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(v);
```
Comment (moran711) [inline modules/contact/contact.model.js:28]: Please move this regexp to config file

File: modules/email-chat/email-chat.model.js [hunk 8]
```diff
@@ -13,7 +14,15 @@ const EmailChatSchema = new mongoose.Schema({
     type: Date,
     default: Date.now,
   },
-  email: String,
+  email: {
+    type: String,
+    validate: {
+      validator: function(v) {
+        return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(v);
```
Comment (moran711) [inline modules/email-chat/email-chat.model.js:21]: same

File: modules/order/order.model.js [hunk 9]
```diff
@@ -20,11 +34,43 @@ const orderSchema = new mongoose.Schema({
...
+    patronymicName: {
+      type: String,
+      minlength: [2, PATRONYMIC_NAME_TO_SHORT],
+      maxlength: [20, PATRONYMIC_NAME_TO_LONG],
+    },
+    email: {
+      type: String,
+      validate: {
+        validator: function(v) {
+          return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(v);
```
Comment (moran711) [inline modules/order/order.model.js:58]: same

File: modules/user/user.model.js [hunk 10]
```diff
@@ -9,15 +18,40 @@ const {
...
+      },
+      message: EMAIL_NOT_VALID,
+    },
+    required: [true, EMAIL_IS_REQUIRED],
+  },
+  phoneNumber: {
+    type: String,
+    validate: {
+      validator: function(v) {
+        return /^\+?3?8?(0\d{9})$/.test(v);
```
Comment (moran711) [inline modules/user/user.model.js:50]: same

File: modules/comment/common/ImageSet.js (hunk 6 above)
Comment (moran711) [inline modules/comment/common/ImageSet.js:23]: You need to set required field for photos

File: modules/order/order.model.js [hunk 11]
```diff
@@ -20,11 +34,43 @@ const orderSchema = new mongoose.Schema({
...
+        },
+        message: EMAIL_NOT_VALID,
+      },
+      required: [true, EMAIL_IS_REQUIRED],
+    },
+    phoneNumber: {
+      type: String,
+      validate: {
+        validator: function(v) {
+          return /^\+?3?8?(0\d{9})$/.test(v);
```
Comment (AndriyLymych) [inline modules/order/order.model.js:68]: Move regexp to constant

File: modules/user/user.model.js [hunk 12]
```diff
@@ -9,15 +18,40 @@ const {
...
     enum: ['user', 'admin', 'superadmin'],
     default: 'user',
   },
-  email: String,
-  phoneNumber: String,
+  email: {
+    type: String,
+    validate: {
+      validator: function(v) {
+        return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(v);
```
Comment (AndriyLymych) [inline modules/user/user.model.js:40]: Same

**Comment by jul4ajka** [comment]
Please remove duplications 13.4%

**Review by jul4ajka — DISMISSED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by AndriyLymych — DISMISSED** [review]

**Review by jul4ajka — APPROVED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by jul4ajka — APPROVED** [review]

**Review by dieie32 — APPROVED** [review]

**Pull request merged**